gcc/rust/hir/rust-hir-dump.cc

```diff
@@ -0,0 +1,446 @@
+// Copyright (C) 2020-2022 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#include "rust-hir-dump.h"
+
+namespace Rust {
+namespace HIR {
+
+Dump::Dump (std::ostream &stream) : stream (stream) {}
+
+void
+Dump::crate (HIR::Crate &crate)
+{
+  // TODO: print crate inner_attrs
+  // TODO: print crate items
+  // TODO: print crate mappings
+  stream << crate.as_string ();
+}
+
+void
+visit (IdentifierExpr &)
+{}
```

Comment:
... like so :) 

However, these do not refer to the `Dump` class's methods as you forgot the `Dump::` prefix. So they are simply functions in that file, without any reference to the class.

I would suggest removing all default definitions in the header (so removing the `{}` and turning them into function declarations, with a semicolon) and adding the `Dump::` prefix on all of those here.

Doing it that way also has the advantage of saving on compilation times, since we will only be changing the source file and not the header whenever modifications are made to your visitor.

You can do this quite easily with two regexes and `sed`, or your editor's equivalent :D 

`sed -i 's/{}/;/g' gcc/rust/hir/rust-hir-dump.h`
`sed -i 's/visit/Dump::&/g' gcc/rust/hir/rust-hir-dump.cc`